src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -0,0 +1,328 @@
+        </ddm:dcmiMetadata>
+      )
+    )
+  }
+
+  "RichElem.setTag" should "report an error" in {
+    import DatasetXml.RichElem
+    val source = new QualifiedSchemedValue[String, String](None, "", "a:b:c")
+    Try {
+      <key>Lorum Ipsum</key>.setTag(DatasetXml.targetFromQualifier, source)
+    } should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "expecting (label) or (prefix:label); got [a:b:c] to adjust the <key> of <key>Lorum Ipsum</key> created from: QualifiedSchemedValue(None,,a:b:c)" =>
+    }
+  }
+
+  "apply" should "report a missing title" in {
+    DatasetXml(minimal.copy(titles = None)) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dc:title" =>
+    }
+  }
+
+  it should "report an empty list of titles" in {
+    DatasetXml(minimal.copy(titles = Some(Seq.empty))) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dc:title" =>
+    }
+  }
+
+  it should "report an empty string as title" in {
+    DatasetXml(minimal.copy(titles = Some(Seq("   \t")))) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dc:title" =>
+    }
+  }
+
+  it should "report empty initials" in {
+    DatasetXml(minimal.copy(creators = Some(Seq(Author(initials = Some(""),surname = Some("")))))) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dcx-dai:initials" =>
+    }
+  }
+
+  it should "report an empty surname" in {
+    DatasetXml(minimal.copy(creators = Some(Seq(Author(initials = Some("x."),surname = Some("")))))) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dcx-dai:surname" =>
+    }
+  }
+
+  it should "report an empty organisation" in {
+    DatasetXml(minimal.copy(creators = Some(Seq(Author(organization = Some("")))))) should matchPattern {
+      case Failure(InvalidDocumentException(_,e)) if e.getMessage ==
+        "no content for mandatory dcx-dai:name" =>
+    }
+  }
+
+  "datasetmetadata.json" should behave like validDatasetMetadata(
+    input = parseTestResource("datasetmetadata.json")
+  )
+  // TODO keep resources in sync with UI module: https://github.com/DANS-KNAW/easy-deposit-ui/blob/784fdc5/src/test/typescript/mockserver/metadata.ts#L246
+  "datasetmetadata-from-ui-some.json" should behave like validDatasetMetadata(
+    input = parseTestResource("datasetmetadata-from-ui-some.json")
+  )
+  "datasetmetadata-from-ui-all.json without one of the authors" should behave like validDatasetMetadata(
+    input = parseTestResource("datasetmetadata-from-ui-all.json"),
+    expectedOutput = Seq(
+      <ddm:profile>
```

Comment:
would it make sense to extract the larger snippets in this class in either their own values or even their own files?

Reply:
I started with this one a kind test driven development of the implementation. In the next iteration I intend to reduce this one to just validate the DDM. Then all variations of individual fields should have smaller tests like the authors and dates.